packages/crypto/src/transactions/serializers/block.ts

```diff
@@ -36,7 +37,20 @@ class BlockSerializer {
     }
 
     private serializeHeader(block: IBlockData, buffer: ByteBuffer): any {
-        block.previousBlockHex = Block.toBytesHex(block.previousBlock);
+        const constants = configManager.getMilestone(block.height - 1);
+
+        if (constants.block.idFullSha256) {
+            if (block.previousBlock.length !== 64) {
+                throw new Error(
```

Comment:
Please create a custom error type (see `errors.ts`) and move all the text there.

Reply:
Done in f7449416b.